code/game/machinery/suit_storage_unit.dm

```diff
@@ -319,8 +319,9 @@
 	MASK = null
 
 /obj/machinery/suit_storage_unit/proc/eject_storage()
-	eject(STORAGE)
-	STORAGE = null
+	if(STORAGE)

```

Comment:
actually that's probably a better location for it, since it will handle if the equipment is missing as well